iree/schemas/cuda_executable_def.fbs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace iree;
+
+// 'Cuda Executable'.
+file_identifier "CUDA";
+file_extension "cuda";
+
+// A struct for the kernel block size along each dimensions.
+struct block_size {
+  x:uint32;
+  y:uint32;
+  z:uint32;
+}
+
+table CudaExecutableDef {
```

Comment:
CUDAExecutableDef* (from #4874 review)

Reply:
review still in progress; comment was:
somewhat of a nit but easier to fix now than later: since nvidia officially uses `CUDA` and we also use all-caps `HAL` it'd be nice if we could consistently call cuda `CUDA` as well - otherwise it looks weird :) should be an easy find-replace

Reply:
Sounds good, thanks for the feedback. I fixed the naming problems.